### PR TITLE
Added popove to owner type field and sharing persons type field [IT451]

### PR DESCRIPTION
**Pull request by areksolek**
Added popove to owner type field and sharing persons type field [IT451]

### CODE REVIEW

File: modules/Vtiger/uitypes/Owner.php [hunk 1]
```diff
@@ -76,23 +76,26 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 				if ($userModel->get('status') === 'Inactive') {
 					$ownerName = '<span class="redColor"><s>' . $ownerName . '</s></span>';
 				}
-				if (App\User::getCurrentUserModel()->isAdmin()) {
-					$detailViewUrl = $userModel->getDetailViewUrl();
+				if ($userModel->get('status') === 'Active' && \App\Privilege::isPermitted('Users', 'DetailView', $value)) {
+					$detailViewUrl = "index.php?module=Users&view=Detail&record={$value}";
+					$popoverRecordClass = 'class="js-popover-tooltip--record"';
+					$dataId = "data-id=\"@$value\" data-js=\"click\"";
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Owner.php]: remove
```suggestion
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 2]
```diff
@@ -90,23 +90,26 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 					if ($userModel->get('status') === 'Inactive') {
 						$ownerName = '<span class="redColor">' . $ownerName . '</span>';
 					}
-					if (App\User::getCurrentUserModel()->isAdmin()) {
-						$detailViewUrl = $userModel->getDetailViewUrl();
+					if ($userModel->get('status') === 'Active' && \App\Privilege::isPermitted('Users', 'DetailView', $value)) {
+						$detailViewUrl = "index.php?module=Users&view=Detail&record={$shownerid}";
+						$popoverRecordClass = 'class="js-popover-tooltip--record"';
+						$dataId = "data-id=\"@$shownerid\" data-js=\"click\"";
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 3]
```diff
@@ -90,23 +90,26 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
...
 					}
 					break;
 				case 'Groups':
 					if (App\User::getCurrentUserModel()->isAdmin()) {
 						$recordModel = new Settings_Groups_Record_Model();
 						$recordModel->set('groupid', $shownerid);
 						$detailViewUrl = $recordModel->getDetailViewUrl();
+						$popoverRecordClass = $dataId = '';
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
						$popoverRecordClass = '';
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 4]
```diff
@@ -90,23 +90,26 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
...
 					}
 					break;
 				default:
 					$ownerName = '<span class="redColor">---</span>';
 					break;
 			}
-			if (!empty($detailViewUrl)) {
-				$displayValue[] = "<a href=\"$detailViewUrl\">$ownerName</a>";
+			if (isset($detailViewUrl)) {
+				$displayValue[] = "<a $popoverRecordClass href=\"$detailViewUrl\" $dataId> $ownerName </a>";
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
				$displayValue[] = "<a $popoverRecordClass href=\"$detailViewUrl\"> $ownerName </a>";
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 5]
```diff
@@ -134,8 +137,10 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
 					if ($userModel->get('status') === 'Inactive') {
 						$shownerData[$key]['inactive'] = true;
 					}
-					if ($isAdmin && !$rawText) {
-						$shownerData[$key]['link'] = $userModel->getDetailViewUrl();
+					if ($userModel->get('status') === 'Active' && (\App\Privilege::isPermitted('Users', 'DetailView', $shownerid) && !$rawText)) {
+						$shownerData[$key]['link'] = "index.php?module=Users&view=Detail&record={$shownerid}";
+						$shownerData[$key]['class'] = 'class="js-popover-tooltip--record"';
+						$shownerData[$key]['data'] = "data-id=\"@$shownerid\" data-js=\"click\"";
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 6]
```diff
@@ -148,8 +153,9 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
 					$detailViewUrl = $recordModel->getDetailViewUrl();
 					if ($isAdmin && !$rawText) {
 						$shownerData[$key]['link'] = $detailViewUrl;
+						$shownerData[$key]['class'] = '';
+						$shownerData[$key]['data'] = '';
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 7]
```diff
@@ -162,7 +168,7 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
 				$shownerName = '<span class="redColor">' . $shownerName . '</span>';
 			}
 			if (isset($shownerData[$key]['link'])) {
-				$shownerName = "<a href='" . $shownerData[$key]['link'] . "'>$shownerName</a>";
+				$shownerName = '<a ' . $shownerData[$key]['class'] . 'href="' . $shownerData[$key]['link'] . '"' . $shownerData[$key]['data'] . '>' . $shownerName . '</a>';
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/SharedOwner.php]:
```suggestion
				$shownerName = '<a ' . $shownerData[$key]['class'] . 'href="' . $shownerData[$key]['link'] . '"' . '>' . $shownerName . '</a>';
```

File: modules/Vtiger/uitypes/Owner.php [hunk 8]
```diff
@@ -76,23 +76,26 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
...
 				}
 				break;
 			case 'Groups':
 				if (App\User::getCurrentUserModel()->isAdmin()) {
 					$recordModel = new Settings_Groups_Record_Model();
 					$recordModel->set('groupid', $value);
 					$detailViewUrl = $recordModel->getDetailViewUrl();
+					$popoverRecordClass = $dataId = '';
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Owner.php]:
```suggestion
					$popoverRecordClass = '';
```

**Comment by rskrzypczak** [comment]
Formating code

File: modules/Vtiger/uitypes/Owner.php [hunk 9]
```diff
@@ -76,23 +76,25 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 				if ($userModel->get('status') === 'Inactive') {
 					$ownerName = '<span class="redColor"><s>' . $ownerName . '</s></span>';
 				}
-				if (App\User::getCurrentUserModel()->isAdmin()) {
-					$detailViewUrl = $userModel->getDetailViewUrl();
+				if ($userModel->get('status') === 'Active' && \App\Privilege::isPermitted('Users', 'DetailView', $value)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/Owner.php]: }elseif (\App\Privilege::isPermitted('Users', 'DetailView', $value)) {

File: modules/Vtiger/uitypes/Owner.php [hunk 10]
```diff
@@ -76,23 +76,25 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
...
+					$detailViewUrl = "index.php?module=Users&view=Detail&record={$value}";
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/Owner.php]:
```suggestion
					$detailViewUrl = 'index.php?module=Users&view=Detail&record='.$value;
```

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 11]
```diff
@@ -90,23 +90,25 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 					if ($userModel->get('status') === 'Inactive') {
 						$ownerName = '<span class="redColor">' . $ownerName . '</span>';
 					}
-					if (App\User::getCurrentUserModel()->isAdmin()) {
-						$detailViewUrl = $userModel->getDetailViewUrl();
+					if ($userModel->get('status') === 'Active' && \App\Privilege::isPermitted('Users', 'DetailView', $value)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/SharedOwner.php]: ?

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 12]
```diff
@@ -90,23 +90,25 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
...
+						$detailViewUrl = "index.php?module=Users&view=Detail&record={$shownerid}";
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/SharedOwner.php]: ?

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 13]
```diff
@@ -134,8 +136,9 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
 					if ($userModel->get('status') === 'Inactive') {
 						$shownerData[$key]['inactive'] = true;
 					}
-					if ($isAdmin && !$rawText) {
-						$shownerData[$key]['link'] = $userModel->getDetailViewUrl();
+					if ($userModel->get('status') === 'Active' && (\App\Privilege::isPermitted('Users', 'DetailView', $shownerid) && !$rawText)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/SharedOwner.php]: ?

File: modules/Vtiger/uitypes/SharedOwner.php [hunk 14]
```diff
@@ -134,8 +136,9 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
...
+						$shownerData[$key]['link'] = "index.php?module=Users&view=Detail&record={$shownerid}";
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/SharedOwner.php]: ?

**Pull request merged**